raster-foundry/raster-foundry
# Better nodata handling in ingest through no nodata handling

## Overview

This PR removes explicitly setting nodata in one place in the ingest pipeline. It helps with bizarre coloring artifacts in Sentinel-2 imagery.

It also includes a revert, and then a revert of that revert, of the change to switch back to ivy resolver from coursier, in case anyone wants to prove that that's the issue.

It also includes a helper function for generating color correction params, because I got tired of typing everything out while I was testing (there are a lot of color correct params, and "just pick some bands" covers the whole non-advanced color correction case).

### Checklist


## Notes

If you want to use backsplash, check out the commit before the revert of the revert, or you'll go insane waiting for logs. Probably just use the existing tile server though.

## Testing Instructions

Here are three scene IDs that you can add to a project that exhibit some of the odd S2 rendering behavior:

```
b994e990-6242-48c7-841e-5281c9cab2a9
2bbf7a2c-c7d8-4a8d-970b-f0141ba4d912
0dba1a1f-aa13-47e0-b80c-4bca0801c79f
```

They're in central Australia.

 * ingest those scenes using the `develop` batch container 
 * add them to a project
 * scroll through zoom levels and observe the coloring artifacts
 * rebuild batch container
 * ingest them again
 * restart memcached
 * scroll through the zoom levels again
 * note that coloring artifacts are gone

Closes the Sentinel-2 part of #4053
